sensor/common/clusterentities/entity_status.go

```diff
@@ -0,0 +1,37 @@
+package clusterentities
+
+func newEntityStatus(numTicks uint16) *entityStatus {
+	return &entityStatus{
+		ticksLeft:    numTicks,
+		isHistorical: false,
+	}
+}
+
+type entityStatus struct {
+	ticksLeft    uint16
+	isHistorical bool
+}
+
+// markHistorical is called when entity would be deleted.
+// Istead, we mark it as historcal and keep it as long as ticksLeft
+func (es *entityStatus) markHistorical(ticksLeft uint16) {
+	if !es.isHistorical {
+		es.ticksLeft = ticksLeft
+	}
+	es.isHistorical = true
+}
+
+// recordTick decreases value of ticksLeft until it reaches 0
+func (es *entityStatus) recordTick() {
+	if !es.isHistorical {
+		return
+	}
+	if es.ticksLeft > 0 {
+		es.ticksLeft--
+	}
+}
+
+// IsExpired returns true if historical entry waited for `ticksLeft` ticks
+func (es *entityStatus) IsExpired() bool {
+	return es.isHistorical && es.ticksLeft == 0
```

Comment:
```suggestion
	return es.isHistorical && es.ticksLeft <= 0
```

Maybe for safety we can check if it is less than 0.

Reply:
`ticksLeft` is uint, so it cannot be smaller than 0, but we can still apply it in case someone changes the type in the future